src/Weft/Server.hs

```diff
@@ -55,8 +52,15 @@ successResponse = responseLBS status200 [(hContentType, "application/json")] . e
 errorResponse :: BL.ByteString -> Response
 errorResponse = responseLBS status500 [(hContentType, "application/json")]
 
-main :: IO ()
-main = do
-    let port = 3000
-    Prelude.putStrLn $ "Listening on port " ++ show port
-    run port $ app gqlResolver -- TODO: allow the user to pass in their resolvers, or Reader them
+server :: (ToJSON (q 'Response), Wefty q) 
+       => [Settings -> Settings]
+       -> Gql q () () 'Resolver
+       -> IO ()
+server s r = runSettings (L.foldl' (&) defaultSettings s) (app r)
```

Comment:
This `[Settings -> Settings]` stuff feels like you're looking for the `Endo` monoid.

`L.foldl' (&) defaultSettings s` could instead be `appEndo (foldMap Endo s) defaultSettings`, or even `ala Endo (($ defaultSettings) . foldMap)` if you're feeling cheeky.

Reply:
@isovector programming with you brings out the cheekiness in me